### PR TITLE
Sensible support for oval 5.11

### PR DESCRIPTION
**Pull request by mpreisler**
#### Description:

- Much simplified the system we need to support both OVAL 5.10 and 5.11
- Some products will have extraneous OVAL defs when built for OVAL 5.10 if those defs are only necessary for OVAL 5.11 to work
- The build_config.yml mechanism is universal and we can use it to pass anything we know at configure time (from cmake or user) to jinja2 templates or even other things. This can enable simpler handling of other arguments, we don't have to keep adding command line arguments to the build scripts.

#### Rationale:

- This was a huge mess and very hard to explain to other people
- Extraneous OVAL defs are not a big deal

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -76,9 +79,12 @@ find_python_module(jinja2 REQUIRED)
...
 
-if (SSG_OVAL_511_ENABLED AND NOT "${OSCAP_V_OUTPUT}" MATCHES "OVAL Version: 5.11")
-    set(SSG_OVAL_511_ENABLED OFF CACHE BOOL "OVAL 5.11 disabled because your version of OpenSCAP doesn't support it" FORCE)
-    message(WARNING "Your version of OpenSCAP does not support OVAL 5.11, disabling OVAL 5.11 for the SSG build.")
+if (SSG_TARGET_OVAL_MAJOR_VERSION EQUAL "5" AND SSG_TARGET_OVAL_VERSION_MINOR EQUAL "11" AND NOT "${OSCAP_V_OUTPUT}" MATCHES "OVAL Version: 5.11")
+    message(FATAL_ERROR "Your version of OpenSCAP does not support OVAL 5.11, please switch the OVAL target version to 5.10 or lower. $ cmake -DSSG_TARGET_OVAL_MINOR_VERSION=10 ../")
+endif()
+
+if (NOT SSG_TARGET_OVAL_VERSION VERSION_EQUAL "5.11" AND NOT SSG_TARGET_OVAL_VERSION VERSION_EQUAL "5.11")
+    message(WARNING "You are targetting OVAL version ${TARGET_OVAL_VERSION}. In SSG we support/test 5.11 and 5.10 only!")
```
Comment (yuumasato) [inline CMakeLists.txt]: It is comparing against "5.11" twice, and  `${TARGET_OVAL_VERSION}` should be `${SSG_TARGET_OVAL_VERSION}`

**Comment by mpreisler** [comment]
(This also fixes SLE11 OVAL 5.10 as a nice little side effect :-))

File: CMakeLists.txt (hunk 1 above)
Comment (mpreisler) [reply CMakeLists.txt]: Good catch, I was inattentive here... Fixed

**Comment by mpreisler** [comment]
@openscap-ci test this please

**Comment by redhatrises** [comment]
@openscap-ci test this please

**Comment by yuumasato** [comment]
@openscap-ci test this please

**Comment by yuumasato** [comment]
@openscap-ci test this please

**Comment by mpreisler** [comment]
This is blocking me so I am using my admin super powers to self-merge... Thx for understading ... :-/

**Pull request merged**